src/stash/src/lib.rs

```diff
@@ -410,7 +410,9 @@ where
         }
     }
 
-    /// Sets the given key value pairs, if not already set.
+    /// Sets the given key value pairs, if not already set. If a new key appears
+    /// multiple times in `entries`, its value will be from the first occurrence
+    /// in `entries`.
```

Comment:
Is this scenario expected? If not we may want to just return an error if any key is duplicated.

Reply:
I'll defer to @mjibson on this detail. As a consumer of this, I'm fine with either this implementation or the error.

Reply:
We're only implementing this because an upcoming PR needs some form of this feature, so yes, it's expected. I think the behavior documented here is defensible because it's exactly what would happen if we instead returned an error (in which case the caller would chunk up their stuff and call this in a loop, where the first key wins).